### PR TITLE
♻️ Require trigger="visibility" to run an <amp-story-animation> effect

### PR DESCRIPTION
**Pull request by alanorozco**
This disambiguates effect specs that trigger with page entrance versus those that are triggered by other specs (subanimations).

Note that subanimations are not currently supported, but this change prevents issues when introducing the feature.

### CODE REVIEW

File: examples/amp-story/amp-story-animation.html [hunk 1]
```diff
@@ -38,7 +38,7 @@
             fly-in with animate-in, after the previous element
           </div>
         </amp-story-grid-layer>
-        <amp-story-animation layout="nodisplay">
```
Comment (alanorozco) [inline examples/amp-story/amp-story-animation.html:41]: `layout` is not necessary since these don't have a custom element implementation.

Comment (dvoytenko) [reply examples/amp-story/amp-story-animation.html:41]: It might be still necessary. Do we want an extra inline element floating in the DOM otherwise?

Comment (alanorozco) [reply examples/amp-story/amp-story-animation.html:41]: I did not realize we had a CSS rule for `[layout=nodisplay]:not(.i-amphtml-element)`. Cool, added it back!

**Comment by gmajoulet** [comment]
Can you elaborate on your plans for the `trigger` parameter on `amp-story-animation`? (This `trigger="in"` is [not documented](https://amp.dev/documentation/components/amp-animation/#trigger-attribute))
What's the API going to look like for subanimations? No trigger at all?

**Comment by dvoytenko** [comment]
> Can you elaborate on your plans for the `trigger` parameter on `amp-story-animation`? (This `trigger="in"` is [not documented](https://amp.dev/documentation/components/amp-animation/#trigger-attribute))
> What's the API going to look like for subanimations? No trigger at all?

Correct. Subanimations have no trigger at all. TBH, I don't like everything about it, but we thought that adding a `trigger` for now would be safer - it will keep the door open for sub-animations done the old way. And if we find a good new way, we can always default it and make it optional. 

The "in" is a new value, just for stories. We can instead go to "visible", which is a standard value.

**Comment by dvoytenko** [comment]
> Sounds good thanks for clarifying :))
> 
> `"visible"` might be easier to understand, or if we want to match the analytics events we send on page navigation for consistency, something like `"story-page-visible"`.

In that case I'd vote for `trigger="visibility"`. It fits ok with the original intent of `amp-animation`.

**Comment by alanorozco** [comment]
Using `visibility`. PTAL

**Pull request merged**